### PR TITLE
v2

### PR DESCRIPTION
**Pull request by vincentvpham**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -85,37 +81,105 @@ export class WildlinkClient {
...
+        ApplicationErrorResponse('WildlinkClient has not been initialized yet'),
+      );
     }
+
     if (!url) {
-      throw new Error('No URL provided');
+      return Promise.reject(ApplicationErrorResponse('No URL provided'));
+    }
+
+    if (url.indexOf(conceptDomain.Value) === -1) {
```
Comment (silushka) [inline src/index.ts]: `url.indexOf(conceptDomain.Value) < 0` is lil bit shorter, but no big difference

File: src/types/api.ts [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-  ok: boolean;
-  json: T;
-  error?: string;
-}
-
-export interface Domain {
+export interface ConceptDomain {
   ID: string;
-  Kind: string;
+  Kind: string[];
```
Comment (silushka) [inline src/types/api.ts]: maybe name `Kinds` could be better

File: test/index.test.ts [hunk 3]
```diff
@@ -4,51 +4,132 @@ import expect from 'expect';
...
+  DeviceToken: DEVICE_TOKEN,
+  DeviceKey: DEVICE_KEY,
+};
+
+describe('WildlinkClient', (): void => {
+  describe('constructor', (): void => {
+    it('should throw if no secret', (): void => {
+      try {
+        new WildlinkClient('', 1);
+        expect((): void => {}).toBeCalledTimes(0);
```
Comment (silushka) [inline test/index.test.ts]: looks like an unnecessary check

Comment (silushka) [reply test/index.test.ts]: will it even reach this step?

File: src/types/api.ts (hunk 2 above)
Comment (vincentvpham) [reply src/types/api.ts]: the name is from our backend (:

File: test/index.test.ts (hunk 3 above)
Comment (vincentvpham) [reply test/index.test.ts]: it will not thats why it is expected to call 0 times. this is meant to catch if the constructor doesnt throw

**Pull request merged**